### PR TITLE
Added project idea from @mr-c and @brainstorm

### PR DESCRIPTION
**Pull request by brainstorm**
As discussed via email, pushing our idea here instead of via the OBF wiki...

### CODE REVIEW

**Comment by peterjc** [comment]
Looks good to me, @kblin shall we leave merges to you at the OBF GSoC 2016 admin?

File: 00_ideas.md [hunk 1]
```diff
@@ -26,3 +25,47 @@ strengths and goals, and are responsible for the quality of the final proposed
...
+It then stores a reference to the code module for the system argparse, and presents the user with all 
+of the functions that stdlib argparse provides. Every function call is passed through the system argparse.
+
+However, gxargparse captures the details of those calls and when a tool definition (Galaxy Tool XML in 
+the case of the original code base) is requested, it builds up the tool definition according to community standards.
+
+Another small proof of concept within the python ecosystem involves the docopt package and 
+[rabixator](https://github.com/stefanristeski/rabixator). Furthermore, other similar instrumentation 
+is being discussed to, for example, explore and optimize the parameter/flag space of i.e, bioinformatic
+aligners with [Teaser](http://biorxiv.org/content/early/2015/09/01/025858).
```
Comment (mr-c) [inline 00_ideas.md]: Nice find!

**Comment by kblin** [comment]
Feel free to merge this. I can't seem to do that myself. Maybe something is still off with the access rights?

File: 00_ideas.md (hunk 1 above)
Comment (mr-c) [inline 00_ideas.md]: Link should be to http://dx.doi.org/10.1186/s13059-015-0803-1

**Comment by peterjc** [comment]
@kblin it looks like your OBF GitHub group membership invitation is still outstanding - can you see this invitation via the website or find the email?

**Comment by hlapp** [comment]
> I can't seem to do that myself. Maybe something is still off with the access rights?

@kblin you need accept invite to the @OBF/document-editors team. It still shows as pending, meaning you haven't accepted it yet.

**Pull request merged**